radix-engine/src/engine/call_frame.rs

```diff
@@ -1067,10 +1072,20 @@ where
         fn_ident: String,
         input: ScryptoValue,
     ) -> Result<ScryptoValue, RuntimeError> {
+        trace!(
+            self,
+            Level::Debug,
+            "Invoking: {:?} {:?}",
+            snode_ref,
+            &fn_ident
+        );
+
         if self.depth == MAX_CALL_DEPTH {
             return Err(RuntimeError::MaxCallDepthLimitReached);
         }
 
+        let is_pay_fee = matches!(snode_ref, SNodeRef::VaultRef(..)) && &fn_ident == "pay_fee";
```

Comment:
I think this should probably be within the `VaultRef` statement rather than outside of the `match` statement?